src/resource/solidDataset.ts

```diff
@@ -518,8 +528,7 @@ export async function saveSolidDatasetInContainer(
     );
   }
 
-  const resourceIri = new URL(locationHeader, new URL(containerUrl).origin)
-    .href;
+  const resourceIri = new URL(locationHeader, response.url).href;
```

Comment:
I think this would be a bug: the URL in `locationHeader` is relative to the origin of the server, rather than the Container's URL, IIRC (but be sure to check):

```suggestion
  const resourceIri = new URL(locationHeader, new URL(response.url).origin).href;
```

Reply:
I actually learnt on Friday that if the relative IRI starts with a `/`, it is relative to the origin of the target URL (that's the case in NSS), and if it starts directly with a path element, it is relative to the complete target URL. Both cases are handled by the `URL` constructor, as well as the case when `locationHeader` is an absolute IRI (as it is the case with ESS), when the `base` should be ignored.

So I think this resolves a potential bug actually :)

Reply:
Weird that I had a test case without a preceding slash before then, but indeed:

> A relative (to the request URL) or absolute URL.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Location

Checking NSS, it doesn't indeed seem to include the leading slash, and ESS returns the entire URL.